wgpu-core/src/command/compute.rs

```diff
@@ -294,37 +320,34 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                     let data_slice =
                         &base.push_constant_data[(values_offset as usize)..values_end_offset];
 
-                    let pipeline_layout = &pipeline_layout_guard[state
+                    let pipeline_layout_id = state
                         .binder
                         .pipeline_layout_id
-                        .expect("Must have a pipeline bound to use push constants")];
+                        .ok_or(ComputePassError::UnboundPipeline)?;
+                    let pipeline_layout = &pipeline_layout_guard[pipeline_layout_id];
 
                     pipeline_layout
                         .validate_push_constant_ranges(
                             wgt::ShaderStage::COMPUTE,
                             offset,
                             end_offset_bytes,
                         )
-                        .unwrap();
+                        .map_err(ComputePassError::from)?;
```

Comment:
You don't actually need this map, the `?` operation should do it for you

Reply:
It does, but here we'd need a double `.into`, so `?` alone won't work.